Bottom: 801d774309e845edb98402c73ff087a78cd2d76f
Top:    dc5f7ffa585b7d6db558fe4437be8a2cf22fd539
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-02-11 16:07:19 +0000

Refresh of ugly-ugly-hack-to-make-pretty-

---

diff --git a/src/draw.c b/src/draw.c
index aa706a5..2a43cd2 100644
--- a/src/draw.c
+++ b/src/draw.c
@@ -2219,33 +2219,25 @@ DrawPlainPolygon (LayerTypePtr Layer, PolygonTypePtr Polygon)
     }
     {
       gui->set_color (Output.fgGC, color);
-#if 0
-      if (!Polygon->NoHolesValid)
-        {
-          ComputeNoHoles (Polygon);
-        }
-      if (Polygon->NoHoles)
-        {
-          PolygonType poly = *Polygon;
-          poly.Clipped = Polygon->NoHoles;
-          do {
-            DrawPolygonFillLowLevel (&poly, NULL);
-            poly.Clipped = poly.Clipped->f;
-          } while (poly.Clipped != Polygon->NoHoles);
-        }
-#endif
-#if 0
+
+      if (gui->fill_pcb_polygon != NULL)
+        gui->fill_pcb_polygon (Output.fgGC, Polygon);
+      else
         {
-          PolygonType poly = *Polygon;
-          poly.Clipped = Polygon->Clipped;
-          do {
-            DrawPolygonLowLevel (&poly, NULL);
-            poly.Clipped = poly.Clipped->f;
-          } while (poly.Clipped != Polygon->Clipped);
+          if (!Polygon->NoHolesValid)
+            {
+              ComputeNoHoles (Polygon);
+            }
+          if (Polygon->NoHoles)
+            {
+              PolygonType poly = *Polygon;
+              poly.Clipped = Polygon->NoHoles;
+              do {
+                DrawPolygonFillLowLevel (&poly, NULL);
+                poly.Clipped = poly.Clipped->f;
+              } while (poly.Clipped != Polygon->NoHoles);
+            }
         }
-#endif
-      gui->fill_pcb_polygon (Output.fgGC, Polygon);
-      //DrawPolygonLowLevel (Polygon, NULL);
 
       /* draw other parts of the polygon if fullpoly flag is set */
       /* NB: No "NoHoles" cache for these */
